addons/product/models/product_product.py

```diff
@@ -64,8 +64,7 @@ class ProductProduct(models.Model):
     product_document_ids = fields.One2many(
         string="Documents",
         comodel_name='product.document',
-        inverse_name='res_id',
-        domain=lambda self: [('res_model', '=', self._name)])
+        inverse_name='res_id')
```

Comment:
So if I have a document on `product.product` 3, it will be shown in the documents of `product.template` 3 as well ? :sparkles:

Reply:
Woopsie

Reply:
Well... Yes, but isn't that the whole point of this task? From what I understand... But the variants are shown with a ruban... So yes, that's what's wanted... oO Did I read the task wrong? xD

Reply:
There should be no modification on the x2many field, it's not shown anywhere.
But the actions should be adapted to a domain:
```
('res_model', '=', 'product.template'), ('res_id', '=', template.id),
OR
('res_model', '=', 'product.product'), ('res_id', 'in', variants.ids),
```